Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/Select.hs

```diff
@@ -121,7 +129,14 @@ instance FromJSON CancelAPIResponse where
   parseJSON err = typeMismatch "Object APISuccess" err
 
 select :: Id DPerson.Person -> Id DEstimate.Estimate -> DSelectReq -> Flow DSelectRes
-select personId estimateId req@DSelectReq {..} = do
+select personId estimateId req = do
+  now <- getCurrentTime
+  selectRes <- select2 personId estimateId req
+  when (selectRes.estimate.validTill < now) $ throwError (InvalidRequest $ "Estimate expired " <> show selectRes.estimate.id) -- select validation check
```

Comment:
This check should come first, before `selectRes <- select2 personId estimateId req`